### PR TITLE
Add /paymentMethods and Generic component

### PR DESCRIPTION
**Pull request by maassenbas**
## Summary
Added /paymentMethods endpoint
Added Generic component as static resource 
Sent /paymentMethods response to frontend and rendered card component

## Tested scenarios

**Fixed issue**:

### CODE REVIEW

**Review by msilvagarcia — APPROVED** [review]

File: force-app/main/default/classes/AdyenClient.cls [hunk 1]
```diff
@@ -35,6 +35,32 @@ public class AdyenClient {
...
+                    HttpRequest httpReq = new HttpRequest();
+                    httpReq.setEndpoint(merchantRecord.PaymentMethods_Endpoint__c);
+                    httpReq.setMethod('POST');
+                    httpReq.setHeader('X-API-KEY', merchantRecord.API_Key__c);
+                    httpReq.setHeader('Content-Type', 'application/json');
+                    String jsonString = JSON.serialize(request, true);
+                    jsonString = Util.stripX(jsonString);
+                    httpReq.setBody(jsonString);
+                    HttpResponse httpResp = httpClient.send(httpReq);
+                    if(httpResp.getStatusCode() == 200) {
```
Comment (cyattilakiss) [inline force-app/main/default/classes/AdyenClient.cls:52]: is 200 the only OK response code from this endpoint that we can receive? In the php lib we have only 1 client and handle a couple of 20x http response codes. 
https://github.com/Adyen/adyen-php-api-library/blob/8.0.0/src/Adyen/HttpClient/CurlClient.php#L130

File: force-app/main/default/classes/AdyenUtil.cls [hunk 2]
```diff
@@ -145,7 +145,18 @@ public with sharing class AdyenUtil {
         
         return request;
     }
-    
+
+    public static OASPaymentMethodsRequest createPaymentMethodsRequest(ccrz__E_Cart__c cart, Merchant__mdt merchant){
+           OASPaymentMethodsRequest paymentMethodsRequest = new OASPaymentMethodsRequest();
+           OASAmount amount = new OASAmount();
+           amount.currency_x = cart.ccrz__CurrencyISOCode__c;
+           amount.value = (Long)cart.ccrz__TotalAmount__c * 100;
```
Comment (cyattilakiss) [inline force-app/main/default/classes/AdyenUtil.cls:153]: Is the *100 always applicable? https://docs.adyen.com/development-resources/currency-codes

File: force-app/main/default/classes/AdyenUtil.cls [hunk 3]
```diff
@@ -145,7 +145,18 @@ public with sharing class AdyenUtil {
...
+           paymentMethodsRequest.amount = amount;
+           paymentMethodsRequest.merchantAccount = merchant.Name__c;
+           paymentMethodsRequest.shopperReference = cart.ccrz__Account__c;
+           paymentMethodsRequest.countryCode = 'US';
```
Comment (cyattilakiss) [inline force-app/main/default/classes/AdyenUtil.cls:157]: I think it's still a TODO, please ignore this if this is noted somewhere to still do

File: force-app/main/default/customMetadata/Merchant.ApexTest.md-meta.xml [hunk 4]
```diff
@@ -22,6 +22,10 @@
         <field>Payment_Endpoint__c</field>
         <value xsi:type="xsd:string">https://checkout-test.adyen.com/v51/payments</value>
     </values>
+    <values>
+        <field>PaymentMethods_Endpoint__c</field>
+        <value xsi:type="xsd:string">https://checkout-test.adyen.com/v51/paymentMethods</value>
```
Comment (cyattilakiss) [inline force-app/main/default/customMetadata/Merchant.ApexTest.md-meta.xml:27]: maybe it's a good idea to go immediately for v64 now that you are building the plugin from the beginning.

File: force-app/main/default/classes/AdyenClient.cls (hunk 1 above)
Comment (maassenbas) [reply force-app/main/default/classes/AdyenClient.cls:52]: Reading the [docs](https://docs.adyen.com/development-resources/response-handling) I assumed it was only 200. Let's discuss and I will update accordingly.

File: force-app/main/default/classes/AdyenUtil.cls (hunk 2 above)
Comment (maassenbas) [reply force-app/main/default/classes/AdyenUtil.cls:153]: In the current package it is indeed always *100. 
Let me create a function for currency check.

File: force-app/main/default/customMetadata/Merchant.ApexTest.md-meta.xml (hunk 4 above)
Comment (maassenbas) [reply force-app/main/default/customMetadata/Merchant.ApexTest.md-meta.xml:27]: Good one. I will update it

**Review by cyattilakiss — APPROVED** [review]

**Pull request merged**